### PR TITLE
Refactor the test framework.

### PR DESCRIPTION
**Pull request by AnashOommen**
1. Move ExampleTestsBase.cs to v2 folder.
2. Add an ExamplesTestSetup.cs file for single-place initialization for the whole test suite.
3. Add an ExampleTestsUtils.cs file with several methods for Setup for tests.

### CODE REVIEW

File: tests/Examples/V2/ExampleTestsUtils.cs [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+        /// <param name="budgetResourceName">The resource name for the budget.</param>
+        /// <returns>The resource name of the newly created campaign.</returns>
+        internal static string CreateCampaign(GoogleAdsClient client, string budgetResourceName)
+        {
+            Campaign campaign = new Campaign()
+            {
+                Name = "Interplanetary Cruise #" + ExampleUtilities.GetRandomString(),
+                AdvertisingChannelType = AdvertisingChannelType.Search,
+
+                // Recommendation: Set the campaign to PAUSED when creating it to prevent
```
Comment (PierrickVoulet) [inline tests/Examples/V2/ExampleTestsUtils.cs]: nit: I would remove unnecessary comments and lines of code as these methods are for util purpose not code examples.

File: tests/Examples/ExampleTestsBase.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2018 Google LLC
```
[the remark this answers is not in the document]
Comment (AnashOommen) [reply tests/Examples/ExampleTestsBase.cs]: This is basically a move of file, not sure why git didn't pick it up correctly.

[the remark this answers is not in the document]
Comment (AnashOommen) [reply tests/Examples/ExampleTestsBase.cs]: This is a moved file, so not changing the copyright text.

File: tests/Examples/ExamplesTestSetup.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Ads.GoogleAds.Util;
+using NUnit.Framework;
+
+namespace Google.Ads.GoogleAds.Tests.Examples
+{
+    [SetUpFixture]
+    public class ExamplesTestSetup
```
[the remark this answers is not in the document]
Comment (AnashOommen) [reply tests/Examples/ExamplesTestSetup.cs]: Done.

File: tests/Examples/V2/ExampleTestsUtils.cs (hunk 1 above)
Comment (AnashOommen) [reply tests/Examples/V2/ExampleTestsUtils.cs]: Done

File: src/Config/GoogleAdsConfig.cs [hunk 4]
```diff
@@ -335,6 +344,19 @@ internal AuthorizationMethod AuthorizationMethod
             set => SetPropertyAndNotify(authorizationMethod, value);
         }
 
+        /// <summary>
+        /// Gets or sets the client customerId.
+        /// </summary>
+        /// <value>
```
[the remark this answers is not in the document]
Comment (AnashOommen) [reply src/Config/GoogleAdsConfig.cs]: Done.

**Pull request merged**